ceno_zkvm/src/scheme/mock_prover.rs

```diff
@@ -1173,35 +1173,6 @@ Hints:
                         let write_rlc_records =
                             filter_mle_by_selector_mle(write_rlc_records, w_selector.clone());
 
-                        if $ram_type == RAMType::GlobalState {
```

Comment:
here is just for better debugging message

Before fix, the debug message during read/write mismatch is 
```
ERROR ceno_zkvm::scheme::mock_prover: riscv_opcode/SW/write_memory/write_record at row 42 (pc=8001a00,ts=832) not found in Memory reads
ERROR ceno_zkvm::scheme::mock_prover: riscv_opcode/SW/write_memory/write_record at row 48 (pc=8001a00,ts=1020) not found in Memory reads
ERROR ceno_zkvm::scheme::mock_prover: riscv_opcode/SW/write_memory/write_record at row 54 (pc=8001a00,ts=1208) not found in Memory reads
```
However, the pc showed here is represent the **NEXT** pc because we generate global state message from state "write" which pc has been bump +4.

The right way is collect this information from global state "read"